### PR TITLE
Admin

### PR DESCRIPTION
**Pull request by dspencer001**

### CODE REVIEW

File: client/js/_admin/actions/applications.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+// Actions that make an api request
+const requests = [
+  'GET_APPLICATIONS',
+  'SAVE_APPLICATION',
+];
+
+export const Constants = wrapper(actions, requests);
+
+export function getApplications() {
```
Comment (jbasdf) [inline client/js/_admin/actions/applications.js]: These should use the arrow function syntax with implicit return

File: client/js/_admin/actions/sites.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const requests = [
+  'GET_SITES',
+  'CREATE_SITE',
+];
+
+export const Constants = wrapper(actions, requests);
+
+export function getSites() {
+  return {
+    type   : Constants.GET_SITES,
```
Comment (jbasdf) [inline client/js/_admin/actions/sites.js]: Be consistent about alignment in POJOS

File: client/js/_admin/components/application_instances/form.jsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+              </label>
+            </div>
+          ))
+        }
+      </div>
+      <h3 className="c-modal__subtitle">Install Settings</h3>
+      <div className="o-grid o-grid__bottom">
+        {
+          _.map(TYPE_RADIOS, (typeLabel, typeValue) => (
+            <div key={typeValue} className="o-grid__item u-third">
```
Comment (jbasdf) [inline client/js/_admin/components/application_instances/form.jsx]: This section should be abstracted into a react component.

File: client/js/_admin/components/common/settings_inputs.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import _     from 'lodash';
+
```
Comment (jbasdf) [inline client/js/_admin/components/common/settings_inputs.jsx]: Add comments describing what this does.

File: client/js/_admin/components/lti_installs/account_install.jsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import React                          from 'react';
+import _                              from 'lodash';
+import {
+  createExternalToolAccounts,
+  deleteExternalToolAccounts,
+ } from '../../../libs/canvas/constants/external_tools';
+
+export default function AccountInstall(props) {
```
Comment (jbasdf) [inline client/js/_admin/components/lti_installs/account_install.jsx]: you should abstract the install function out to a parent class and then pass it in as a prop.

File: client/js/_admin/components/lti_installs/course_install_row.jsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import React                     from 'react';
+import {
+  createExternalToolCourses,
+  deleteExternalToolCourses,
+} from '../../../libs/canvas/constants/external_tools';
+
+export default function CourseInstallRow(props) {
```
Comment (jbasdf) [inline client/js/_admin/components/lti_installs/course_install_row.jsx]: same here abstract out functions and put them in parent class and pass as prop, or make this a class if the parent is becoming unwieldy

File: client/js/_admin/components/lti_installs/index.jsx [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+          />
+        </div>
+        <ReactModal
+          isOpen={this.props.loadingAccounts}
+          contentLabel="Modal"
+          overlayClassName="c-modal__background"
+          className="c-modal c-modal--newsite is-open c-modal--error loading"
+        >
+          <div className="c-loading-icon" />
+          &nbsp;&nbsp;&nbsp;Loading...
```
Comment (jbasdf) [inline client/js/_admin/components/lti_installs/index.jsx]: is there really not a better way to do this via padding or some other mechanism?

File: client/js/_admin/components/lti_installs/sidebar.jsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+import SubAccounts from './sub_accounts';
+
+export default function Sidebar(props) {
+  const schoolName = props.accounts[1] ? props.accounts[1].name : 'Loading...';
+  const subAccounts = props.accounts[1] ? props.accounts[1].sub_accounts : [];
+
+  return (
+    <div className="o-left">
+      <div className="c-tool">
+        <a href=""><i className="i-settings" /></a>
```
Comment (jbasdf) [inline client/js/_admin/components/lti_installs/sidebar.jsx]: is this supossed to do something?

File: client/js/_admin/components/sites/form.jsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  oauth_secret : 'Canvas Developer Key',
+  url          : 'Canvas Domain',
+};
+
+export default function NewSiteForm(props) {
+  return (
+    <form>
+      <div className="o-grid o-grid__modal-top">
+        {
+          _.map(FIELDS, (fieldLabel, field) => (
```
Comment (jbasdf) [inline client/js/_admin/components/sites/form.jsx]: same here, lets abstract these things into component

File: client/js/_admin/reducers/accounts.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    case Constants.GET_CANVAS_ACCOUNTS: {
+      const newState = _.cloneDeep(state);
+      newState.loading = _.isEmpty(newState.accounts);
+
+      return newState;
+    }
+
+    case 'GET_SUB_ACCOUNTS_OF_ACCOUNT_DONE': {
```
Comment (jbasdf) [inline client/js/_admin/reducers/accounts.js]: this doesn't modify the state in any way so it probably doesn't work like intended

Comment (dspencer001) [inline client/js/_admin/reducers/accounts.js]: Accounts and subaccounts are stored nested. This is modifying the original account passed in to put the subaccounts in the right place. A shallow clone to tell any components that it is a new state an maintain object references would probably be good though.

**Pull request merged**